presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -78,6 +79,10 @@
     // add one extra entry for null after stripe/rowGroup dictionary
     private static final int[] EMPTY_DICTIONARY_OFFSETS = new int[2];
 
+    // Each rowgroup has roughly 10K rows, and each batch reads 1K rows. So there're about 10 batches in a rowgroup.
+    private static final int BATCHES_PER_ROWGROUP = 10;
+    private static final float MATERIALIZATION_RATIO = 2.0f;
+
```

Comment:
Could you elaborate on where the numbers came from? Is this something that's standard and well known?

Reply:
https://orc.apache.org/specification/ORCv1/ states that the default row group has 10,000 rows. Since each batch has at most 1024 rows, the BATCHES_PER_ROWGROUP is roughly 10.
MATERIALIZATION_RATIO should be a value greater than 1.0f to account for extra CPU required to materialize the VariableWidthBlock. I picked 2.0f here which is a safe choice, so that it doesn't over-radically materialize blocks. This value can be adjusted in the future, but I think it's over-kill to introduce configuration properties because this is too technical and users won't know how to configure it.

Reply:
I added the following comments
`// MATERIALIZATION_RATIO should be greater than or equal to 1.0f to compensate the extra CPU to materialize blocks.`